docs/details/signal.dox

```diff
@@ -138,31 +146,38 @@ The last entry in the table has more permutations than shown here.
 
 \copydoc signal_func_conv_desc
 
-For two dimensional signals, the following are the possible batch operations
-possible. Lets say m & n as sizes along the 0th & 1st dimensions respectively
-and p & q are the some integral numbers greater than one.
+For two dimensional signals, lets assume \f$[M\quad N]\f$, \f$[A\quad B]\f$ are
+signal, filter sizes along \f$0^{th}\f$ & \f$1^{st}\f$ axes respectively.
+Given two integral numbers greater than one, \f$p\f$ & \f$q\f$, lets take a look
+at the possible batch operations in the below table.
+
+| Signal Size    | Filter Size    | Output Size    | Batch Mode      | Description                                                    |
+| :------------: | :-------------:| :------------: | :-------------: | :------------------------------------------------------------- |
+| \dims{M,N,1,1} | \dims{A,B,1,1} | \dims{M,N,1,1} | No Batch        | \shape_eq{O,M,N,1,1} = \convolve_eq{\shape_t{I,M,N,1,1},\shape_t{f,A,B,1,1}} |
```

Comment:
If we go ahead with the tables:
- Can we define O, I, and f somewhere to rule out ambiguity?
- Can we define the notation we use here (array name above and bracketed dims below), since it's non-standard among literature?
- nit: The bar traditionally means "such that", so it's kind of awkward to read something like "...such that for all p in P" (the second case, for example). Also, traditionally, what follows after the "in" sign is a set, but P seems like a single number, so someone who tries to read this rather rigorously, might be confused to read something like "for all p in 200". How about this: "... | p < P" ?
- Some parts of the notation can be inconsistent or ambiguous. For example, consider the second case again, where "f" has "[A B p 1]" below it. Up to this point in the table, the [ A B C D ] notation is used to say that the array f has dim0 = A, dim1 = B, dim2 = C, etc, but here, "p" seems to be used as an index in [0, P), and not the dim size. It could be ambiguous whether the numbers inside [ ] are dims or indices. Consequently, does it mean that I*f inside {} are slices, and the set {} itself is the whole output array?
- In light of what I said above, I'm not sure what to do here, but maybe it'll help if we describe how the notation works. Maybe even give an example usage of the notation and describe that example in words.

Reply:
* I don't think you caught this but I defined p and q instead of P and Q before the table. I have refactored those variables there.
* I have refactored O and I with actual words, so there is not new symbol now.
* I am not sure I  agree with that. A lot of papers use the underset notation to describe a tensor/matrix with it's size under the tensor's name.
* We can use both `:` and `|` in set builder notation. The reason I chose `|` is that it visually helps it separating the left hand term from the right hand criteria.
* I am looking into it changing the p and q portions, to something like range which should help the case.
* I don't think I have used A B C D anywhere, the sentence before the table only establishes the size of signal and filter along 0th and 1st axes. Yeah, as I said in the first item in this list, I have fixed that discrepancy of p and P.
* About numbers inside `[ ]` in description column, once I fix the p notation issue this will be fixed automatically.

I don't think we need to explain what set builder notation is again in our documentation as long as we fix above problems and stick to basic notations without ambiguity.